File: solutions/elixir/bob/1/lib/bob.ex
defmodule Bob do
  @spec hey(String.t()) :: String.t()
  def hey(input) do
    input = input |> String.trim()

    cond do
      input |> String.length() == 0 -> "Fine. Be that way!"
      question?(input) and shout?(input) -> "Calm down, I know what I'm doing!"
      question?(input) -> "Sure."
      shout?(input) -> "Whoa, chill out!"
      true -> "Whatever."
    end
  end

  defp question?(input), do: input |> String.ends_with?("?")
  defp shout?(input), do: !(input =~ ~r/[[:lower:]]/) && input =~ ~r/[[:upper:]]/
end

File: solutions/clojure/bob/1/src/bob.clj
(ns bob
  (:require [clojure.string :as str]))

(defn response-for [s]
  (let [trimmed (str/trim s)
        question? (str/ends-with? trimmed "?")
        shout? (and (not= trimmed (str/lower-case trimmed))
                    (= trimmed (str/upper-case trimmed)))]
    (cond
      (empty? trimmed) "Fine. Be that way!"
      (and question? shout?) "Calm down, I know what I'm doing!"
      question? "Sure."
      shout? "Whoa, chill out!"
      :else "Whatever.")))
